### ISSUE TITLE
inconvience with rebounding trace()

### ISSUE THREAD

**Issue by oprudkyi** [HaxeFoundation/haxe#5264]
it seems like Log:trace() does two things at once -  process 'infos' parameter and output trace message, 
https://github.com/HaxeFoundation/haxe/blob/development/std/haxe/Log.hx

in real scenarios rebounding trace ends with relatively ugly solution - you have to reimplement all logic just to point trace into other place 

probably will be better to have additional function, some like Log.outputTrace just for simple string-only output , so rebounding trace won't look ugly 
see bottom of https://github.com/oprudkyi/thrift/blob/d130c35f9d0be9ae3b919205d0698492243fb51c/lib/haxe/README.md

this issue based on conversation at the https://github.com/apache/thrift/pull/1013

**Comment by nadako** [comment]
Agreed, however this change would probably be a bit too big. Alternatively we could add some `formatTrace` method that does the processing and returns a string. It would be used by default and can be reused in custom `trace` functions.
